spec/unit/docsettings_spec.lua

```diff
@@ -143,4 +143,65 @@ describe("docsettings module", function()
         assert.Equals("file", lfs.attributes(d.sidecar_file, "mode"))
         assert.Equals("file", lfs.attributes(d.sidecar_file .. ".old", "mode"))
     end)
+
+    it("should ignore empty file", function()
+        local file = "file.pdf"
+        local d = docsettings:open(file)
+        d:saveSetting("a", "a")
+        d:flush()
+        -- metadata.pdf.lua should be generated.
+        assert.Equals("file", lfs.attributes(d.sidecar_file, "mode"))
+        d:close()
+        -- metadata.pdf.lua and metadata.pdf.lua.old should be generated.
+        assert.Equals("file", lfs.attributes(d.sidecar_file, "mode"))
+        assert.Equals("file", lfs.attributes(d.sidecar_file .. ".old", "mode"))
+
+        -- reset the sidecar_file to an empty file.
+        local f_out = io.open(d.sidecar_file, "w")
+        f_out:close()
+
+        d = docsettings:open(file)
+        -- metadata.pdf.lua should be removed.
```

Comment:
Not an objection to merging, but for clarity it could be an idea to write the test more like:

```lua
describe("ignore empty file", function() -- sub-describe block
  local file, etc
  it("should generate metadata.pdf.lua", function()
    assert.Equals()
  end)
  it("should generate metadata.pdf.lua and metadata.pdf", function()
    assert.Equals()
  end)
end)
```

That way the comments become an explicit part of the test suite.

Reply:
Oh, I did not know that.